File: CppPractice/CppLeetcode/Leetcode121.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int ans=0;
        int curLow = prices[0];
        for(int i=1;i<prices.size();i++)
        {
            if(prices[i]<curLow){curLow=prices[i];}
            ans=max(ans,prices[i]-curLow);

            
        }
        return ans;
        
    }
};

File: CSharpPractice/LeetcodePractice/Leetcode121.cs
public class Solution121 {
    public int MaxProfit(int[] prices) 
    {
        int ans = 0;
        int minPrice = int.MaxValue;
        int curPrice = 0;
        int temp=0;
        foreach(int price in prices)
        {
            minPrice = Math.Min(price,minPrice);
            temp = price - minPrice;
            ans = Math.Max(temp,ans);
        }
        return ans;
    }
}